Clean up code & improve test coverage

ondras12345 opened the pull request:
This PR tries to clean up the code by:
- using f-strings instead of old %-style format strings and `.format()`,
- using pathlib.Path and `.as_posix()` instead of `os.path.*` and manually replacing `\` with `/`,
- dropping legacy code that was needed for Python2 compatibility,
- dropping unreachable code.

There should be no significant changes in behavior (except for the last commit "Fix createCSVData on Windows", which removes an extra newline at the end of the csv file).

It also adds tests, improving `OMPython/__init__.py` coverage to 73% on Linux.

syntron commented on OMPython/__init__.py:
```diff
@@ -279,7 +279,7 @@ def _set_omc_command(self, omc_path_and_args_list):
             self._omc_command = omc_path_and_args_list
         else:
-            self._omc_command = ' '.join([shlex.quote(a) if (sys.version_info > (3, 0)) else a for a in omc_path_and_args_list])
+            self._omc_command = ' '.join([shlex.quote(a) for a in omc_path_and_args_list])
```
would this work also without the join()? - it is just the command used for subprocess - this works well as a list; we only have to check that all items are quoted

ondras12345 replied:
I have just tried it, and it doesn't seem to like the change:
```
/home/ondra/source/repos/OMPython/tests/test_ZMQ.py:17: in __init__
    self.om = OMPython.OMCSessionZMQ()
/home/ondra/source/repos/OMPython/OMPython/__init__.py:551: in __init__
    self._connect_to_omc(timeout)
/home/ondra/source/repos/OMPython/OMPython/__init__.py:581: in _connect_to_omc
    raise Exception(f"OMC Server did not start (timeout={timeout}). Could not open file {self._port_file}"
)
E   Exception: OMC Server did not start (timeout=10.0). Could not open file /tmp/openmodelica.ondra.port.c431836ad07c4d41b06cd1eee41cffff
```

However, I don't know why. At first glance, it should work. E.g. this works:
```
In [1]: import subprocess

In [2]: subprocess.Popen(['/usr/bin/omc', '--help'])
Out[2]: <Popen: returncode: None args: ['/usr/bin/omc', '--help']>

OpenModelica Compiler OpenModelica 1.25.0
Copyright © 2019 Open Source Modelica Consortium (OSMC)
Distributed under OMSC-PL and GPL, see www.openmodelica.org

Usage: omc [Options] (Model.mo | Script.mos) [Libraries | .mo-files] 
[...]
```

I will investigate further tomorrow.

ondras12345 replied:
Fixed in [20c24dc](https://github.com/OpenModelica/OMPython/pull/249/commits/20c24dc666be595608246f8d2e200e84e2a0c9bb)
The reason it didn't work was that Popen can't take a list with shell=True.

syntron commented:
@ondras12345 I realy like the work you have done; especially the string formating! I was thinking about this myself but stopped as I was not sure how much of it is needed for Python 2.x compatibility. However, as this is droped, these changes make the code more readable!

ondras12345 commented:
> how much of it is needed for Python 2.x compatibility. However, as this is droped

I didn't even notice the README said the library still supports Python2, and since setup.py said `python_requires='>=3.8'`, I assumed that's the oldest version we need to support and started removing python2 workarounds in https://github.com/OpenModelica/OMPython/pull/246.

Also as a side note, if support for 3.8 is declared in `pyproject.toml`, we should probably run tests @ 3.8 and not just 3.10 and 3.12:
https://github.com/OpenModelica/OMPython/blob/bfb115376fa4090592566fd8d4dfafd2dc27a238/.github/workflows/Test.yml#L15
To be fair, I didn't test my changes with Python 3.8. I should go do that tomorrow...

ondras12345 commented:
I tried to run a set of tests with python 3.8, and it doesn't even install. And it's totally my fault, because it fails to parse pyproject.toml, which was introduced by me in https://github.com/OpenModelica/OMPython/pull/246. Error log:
https://github.com/ondras12345/OMPython/actions/runs/14511691030/job/40711663255
I'll see what I can do about it.

syntron commented:
> and they are passing. It should be safe to say python >= 3.10 is supported. It's not like we need the new features introduced in 3.11 and 3.12.

I support staying with Python >= 3.10 as this version is working well in the tests; why should we limit the possible user range without a really need for such a change?

ondras12345 commented:
I have rebased my branch onto master and fixed the conflicts (there was a lot of them).
If the tests pass (especially the windows ones, I already ran linux tests locally), this PR should be ready for review @adeas31

arun3688 reviewed (approved):
looks good

adeas31 reviewed (approved):
LGTM. Thanks!
